test/e2e/backup_restore_test.go

```diff
@@ -53,6 +53,10 @@ func testBackupRestore(clusterSpec clusterSpec) func(*testing.T) {
 
 		// 3. Backup the cluster's content
 		backupSpec := &v2.Backup{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "infinispan.org/v2alpha1",
+				Kind:       "Backup",
+			},
```

Comment:
Why is adding the `TypeMeta` necessary?

Reply:
GVK requires on OCP/OKD for Update. Probably not in this case, but we should don't forget about it
Please have a look at this case and code
https://github.com/infinispan/infinispan-operator/blob/f44347dfd4c5484425a75b11ddde07f0b5fda941/test/e2e/main_test.go#L706-L708